File: 1 Algoritmos clásicos de optimización/Códigos/Newton Raphson/plot_2D.py
import matplotlib.pyplot as plt
import numpy as np

def plot_2d(f,x,xl,xu):
    plt.clf()
    X = np.arange(xl, xu, 0.25)
    Y = f(X)

    plt.plot(X,Y)
    plt.plot(x, f(x), marker="o", c='r', markersize=10)

    plt.title("Plot")
    plt.xlabel('x')
    plt.ylabel('f(x)')

    plt.show(block=False)
    plt.pause(.05)

File: 1 Algoritmos clásicos de optimización/Códigos/Newton Raphson/Plot_2D.m
function Plot_2D (f,x,xl,xu)
    cla
    hold on
    grid on
    
    xp = xl:0.01:xu;
    plot(xp,f(xp),'b-','LineWidth',3,'MarkerSize',12)
	plot(x,f(x),'*r','LineWidth',3,'MarkerSize',12);

    xlabel('x')
    ylabel('f(x)')
    
    pause(0.5)
    